mobile/library/common/extensions/filters/http/network_configuration/filter.cc

```diff
@@ -64,19 +96,70 @@ NetworkConfigurationFilter::decodeHeaders(Http::RequestHeaderMap& request_header
     return Http::FilterHeadersStatus::Continue;
   }
 
-  // If there is no proxy configured, continue.
+  // TODO(Augustyniak): Update Android proxy resolution to use API extension registry. As of now,
+  // it's only iOS that uses that code path.
+  const auto proxy_resolver =
+      static_cast<envoy_proxy_resolver*>(Api::External::retrieveApiSafe("envoy_proxy_resolver"));
+  if (proxy_resolver != nullptr) {
+    return resolveProxy(request_headers, proxy_resolver);
+  }
+
+  // TODO(Augustyniak): Migrate Android so that it uses API registry instead of calling
+  // getProxySettings().
   const auto proxy_settings = connectivity_manager_->getProxySettings();
+  return continueWithProxySettings(proxy_settings);
+}
+
+Http::FilterHeadersStatus
+NetworkConfigurationFilter::resolveProxy(Http::RequestHeaderMap& request_headers,
+                                         envoy_proxy_resolver* proxy_resolver) {
+
+  const auto url_string = Http::Utility::buildOriginalUri(request_headers, absl::nullopt);
+  const auto host_data = Data::Utility::copyToBridgeData(url_string);
+
+  envoy_proxy_resolver_proxy_resolution_result_handler* result_handler =
+      static_cast<envoy_proxy_resolver_proxy_resolution_result_handler*>(
+          safe_malloc(sizeof(envoy_proxy_resolver_proxy_resolution_result_handler)));
+
+  auto weak_self = weak_from_this();
+  auto wrapper =
+      new std::unique_ptr<std::weak_ptr<NetworkConfigurationFilter>>(new std::weak_ptr(weak_self));
+  result_handler->context = static_cast<const void*>(wrapper);
+  result_handler->proxy_resolution_completed = proxyResolutionCompleted;
+
+  envoy_proxy_settings_list proxy_settings_list;
+
+  const auto proxy_resolution_result = proxy_resolver->resolve(
```

Comment:
why do we need to call this to get proxy_settings instead of going through the ConnectivityManager?

Reply:
connectivity manager seems to only handle host/port, not a PAC file: https://github.com/envoyproxy/envoy/blob/be4b2668f6ddb01b2e289f6853572d3a30d15797/mobile/library/common/main_interface.cc#L84-L89